yb-voyager/cmd/exportData.go

```diff
@@ -735,50 +742,99 @@ func reportUnsupportedTables(finalTableList []sqlname.NameTuple) {
+	return tuple, nil
+}
+
+func applyTableListFlagsOnFullListAndAddLeafPartitions(fullTableList []sqlname.NameTuple, tableListViaFlag string, excludeTableListViaFlag string) ([]sqlname.NameTuple, error) {
+	var err error
+	var includeTableList, excludeTableList []sqlname.NameTuple
+
+	applyFilterAndAddLeafTable := func(flagList string, flagName string) ([]sqlname.NameTuple, error) {
+		flagTableList := extractTableListFromString(fullTableList, flagList, flagName)
+		_, flagTableList, err = addLeafPartitionsInTableList(flagTableList, true)
 		if err != nil {
-			utils.ErrExit("adding leaf partititons to exclude table list: %s", err)
+			return nil, fmt.Errorf("failed to add the leaf partitions in %s table list: %w", flagName, err)
 		}
+		return flagTableList, nil
 	}
-	if source.TableList != "" {
-		tableList = extractTableListFromString(fullTableList, source.TableList, "include")
-	} else {
-		tableList = fullTableList
+
+	if excludeTableListViaFlag != "" {
+		//Apply exclude table list filter if present
+		excludeTableList, err = applyFilterAndAddLeafTable(excludeTableListViaFlag, "exclude")
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	includeTableList = fullTableList
+	if tableListViaFlag != "" {
+		//Apply include table list filter if present
+		includeTableList, err = applyFilterAndAddLeafTable(tableListViaFlag, "include")
+		if err != nil {
+			return nil, err
+		}
+	}
+	//return the final table list generated from the command flags  table-list (include list) / exclude-table-list in this run
+	return sqlname.SetDifferenceNameTuples(includeTableList, excludeTableList), nil
+}
+
+func fetchTablesNamesFromSourceAndFilterTableList() (map[string]string, []sqlname.NameTuple, error) {
+	var tableListInFirstRun []sqlname.NameTuple
+	var nameTupleTableListFromDB []sqlname.NameTuple
+	var err error
+	tableListFromDB := source.DB().GetAllTableNames()
+	for _, t := range tableListFromDB {
+		tuple, err := getNameTupleFromQualifiedObject(t.Qualified.Quoted, nil, false)
+		if err != nil {
+			return nil, nil, fmt.Errorf("error getting name tuple for the object: %s: %v", t.Qualified.Quoted, err)
+		}
+		nameTupleTableListFromDB = append(nameTupleTableListFromDB, tuple)
 	}
-	finalTableList = sqlname.SetDifferenceNameTuples(tableList, excludeTableList)
-	isTableListModified := len(sqlname.SetDifferenceNameTuples(fullTableList, finalTableList)) != 0
+
+	//apply table list flags filter on the nameTupleTableListFromDB
+	tableListInFirstRun, err = applyTableListFlagsOnFullListAndAddLeafPartitions(nameTupleTableListFromDB, source.TableList, source.ExcludeTableList)
+	if err != nil {
+		return nil, nil, fmt.Errorf("error applying table list flags on full table list: %v", err)
+	}
+
+	isTableListModified := len(sqlname.SetDifferenceNameTuples(nameTupleTableListFromDB, tableListInFirstRun)) != 0
 	if exporterRole == SOURCE_DB_EXPORTER_ROLE {
 		metaDB.UpdateMigrationStatusRecord(func(record *metadb.MigrationStatusRecord) {
 			if isTableListModified {
```

Comment:
unrelated to your changes, but do we need to set this differently for export-data-from-source/target?

Reply:
Nope, this is mostly used for offline migration and display purposes only